### PR TITLE
fix(std/node): small fixups in `events` and `util` modules

### LINKED ISSUE #5397: std/node: remove util.validateIntegerRange
`std/node/util.ts` exports `validateIntegerRange`. This isn't a function in Node as far as I can tell, and looks like it should be a _private_ utility.

If that's the case, I can move the function to `src/node/_utils.ts` file to fix this. If there are any concerns about this being a breaking change, lemme know!

### PR DESCRIPTION
**Pull request by schwarzkopfb**
In Node it is allowed to set the max event listener count to `Infinity` to disable "possible memory-leak" warnings (setting it to `0` has the same effect). Now this is also possible in Deno's compatibility layer.

Don't expose `util.validateIntegerRange()` as it's not part of Node's API, moved it to internal utilities. Closes #5397

### CODE REVIEW

File: std/testing/asserts.ts [hunk 1]
```diff
@@ -412,7 +412,7 @@ export function assertMatch(
 /**
  * Forcefully throws a failed assertion
  */
-export function fail(msg?: string): void {
```
Comment (bartlomieju) [inline std/testing/asserts.ts:415]: Seems like unrelated change. Please revert

Comment (schwarzkopfb) [reply std/testing/asserts.ts]: You're right, thanks! Reverted.

**Comment by bartlomieju** [comment]
@schwarzkopfb LGTM, could you merge with `master` branch?

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @schwarzkopfb

**Pull request merged**